Makefile

```diff
@@ -1,7 +1,16 @@
-clean:
+clean: nondocs-clean docs-clean
```

Comment:
I wonder whether `clean` should clean everything (including Python caches).

Reply:
I am ambivalent about this one because it you don't use docker it will introduce some slowness every time you run `make docs` or `make it` as I've noticed some downloads happening. At the same time it ensures consistency.

Reply:
Ok, it does not make sense to clean Python caches for `make docs`. However, I think for `make it` would make sense?

Reply:
Yes it would make sense for `make it`. I think we should add it to the individual `it34/35/36` targets too?

Reply:
Yes, please.